src/ServiceControl.Monitoring/Metrics/Raw/LongValueOccurrences.cs

```diff
@@ -1,23 +1,68 @@
 ﻿namespace ServiceControl.Monitoring.Metrics.Raw
 {
+    using System;
+    using System.Collections.Concurrent;
     using NServiceBus;
 
     public class LongValueOccurrences : IMessage
     {
-        public long Version;
+        public class Pool
+        {
+            public static readonly Pool Default = new Pool();
+            ConcurrentStack<LongValueOccurrences> pool = new ConcurrentStack<LongValueOccurrences>();
 
-        public long BaseTicks;
+            public LongValueOccurrences Lease()
+            {
+                LongValueOccurrences value;
+                if (pool.TryPop(out value))
+                {
+                    return value;
+                }
 
-        public int[] Ticks;
+                return new LongValueOccurrences();
```

Comment:
This means that there is no upper bound on the pool size right? I guess we should be fine as this will be throttled by concurrency limit on the endpoint itself.